ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -75,7 +77,7 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 	var err error
 	var rayJobInstance *rayv1alpha1.RayJob
 	if rayJobInstance, err = r.getRayJobInstance(ctx, request); err != nil {
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+		return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, client.IgnoreNotFound(err)
```

Comment:
In this case, if we delete a RayJob, will it be reconciled indefinitely?

Reply:
Yes, based on the discussion on this PR https://github.com/ray-project/kuberay/pull/850#issuecomment-1368094367 it sounds like we are always supposed to requeue, but let me know if the situation for RayJob is different for some reason.  I noticed that without this requeueing, there was a delay of a few minutes before a completed job status was reflected correctly.

Reply:
Do you recommend that the RayJobController watch the k8s job somehow so that a reconciliation is automatically triggered when the k8s job completes? Is this something we need to add or would it happen automatically?

Reply:
> In this case, if we delete a RayJob, will it be reconciled indefinitely?

I conducted an experiment in #1196. The RayJob resource event will continue to be reconciled indefinitely, even if the RayJob custom resource is deleted.

Reply:
@kevin85421 Thanks for running the test!  In this case, is the fix to change this line to `ctrl.Result{}`? Do we need to change back any of the other occurrences as well?

Reply:
We can refer to the following example provided by the operator-framework.

* https://sdk.operatorframework.io/docs/building-operators/golang/advanced-topics/
* https://github.com/operator-framework/operator-sdk/blob/b794fe909abc1affa1f28cfb75ceaf3bf79187e6/testdata/go/v3/memcached-operator/controllers/memcached_controller.go#L88-L100

Reply:
Thanks for the suggestion! Fixed 2bac395

Reply:
Oh, I guess that on line 99 of `memcached_controller.go`, ctrl.Result{RequeueAfter: ...} should be used instead. The documentation may be outdated, so the changes made in https://github.com/ray-project/kuberay/commit/2bac395595d89ac53caa9e8fb2d1090be20e8537 may not make sense.

https://github.com/operator-framework/operator-sdk/commit/1ad6ff3db60d696bb0dd07553c90947d9bdc828d#diff-50e545e0f0125576d93a2a8f95fafcae3cc3aed35a6c8ad93c723834b809d940